src/entropymode.rs

```diff
@@ -0,0 +1,703 @@
+// Copyright (c) 2018, The rav1e contributors. All rights reserved
+//
+// This source code is subject to the terms of the BSD 2 Clause License and
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+#![allow(non_upper_case_globals)]
+
+use context::*;
+use partition::*;
+
+const PALATTE_BSIZE_CTXS: usize = 7;
+const PALETTE_Y_MODE_CONTEXTS: usize = 3;
+const PALETTE_UV_MODE_CONTEXTS: usize = 2;
+const PALETTE_COLOR_INDEX_CONTEXTS: usize = 5;
+const CDFMAX : u16 = 32768; 
+const BLOCK_SIZE_GROUPS: usize = 4;
+const RESTORE_SWITCHABLE_TYPES: usize = 3;
+const TX_SIZE_CONTEXTS: usize = 3;
+
+// from seg_common.h
+const MAX_SEGMENTS: usize = 8;
+const SPATIAL_PREDICTION_PROBS: usize = 3;
+const SEG_TEMPORAL_PRED_CTXS: usize = 3;
+
+// enums.h
+const TX_SIZE_LUMA_MIN:usize = TxSize::TX_4X4 as usize;
+const TX_SIZE_CTX_MIN: usize = (TX_SIZE_LUMA_MIN + 1);
+const MAX_TX_CATS: usize = (TxSize::TX_SIZES - TX_SIZE_CTX_MIN);
+const MAX_TX_DEPTH: usize = 2;
+
+// LUTS ---------------------
+
+// The spec indicates that BLOCK_SIZES should be 22, but BLOCK_SIZES_ALL is defined as 24 in the rust code.
+// There are also two additional symbols that do not appear in the spec: BLOCK_128X32 and BLOCK_32X128.
+
+const BLOCK_SIZES:usize = BlockSize::BLOCK_SIZES_ALL - 2;
```

Comment:
OK. I will raise an issue for that since it's probably out of scope for this PR.

Reply:
OK. Even the EXT_TX_SIZES rename could be made separate, though it's pretty simple to include in this patch.

Reply:
Fair enough, that one is pretty straightforward though. Just renaming a variable in 6 or 7 spots in the code (most of them being in entropymode.rs).  I took a quick try at the BLOCK_SIZES_ALL change and it would be a bit more involved - there are some arrays and things that rely on the value being 24, for example.